lightning/src/ln/peer_handler.rs

```diff
@@ -864,7 +873,20 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 									peer.their_node_id = Some(their_node_id);
 									insert_node_id!();
 									let features = InitFeatures::known();
-									let resp = msgs::Init { features };
+									let resp = msgs::Init { features, remote_network_address:  match peer.their_net_address.clone(){
+										Some(NetAddress::IPv4{addr: [10, 0..=255, ..], port: _}) => None,
+										Some(NetAddress::IPv4{addr: [0, 0..=255, ..], port: _}) => None,
+										Some(NetAddress::IPv4{addr: [100, 64..=127, ..], port: _}) => None,
+										Some(NetAddress::IPv4{addr: [127, 0..=255, ..], port: _}) => None,
+										Some(NetAddress::IPv4{addr: [169, 254, ..], port: _}) => None,
+										Some(NetAddress::IPv4{addr: [172, 16..=31, ..], port: _}) => None,
+										Some(NetAddress::IPv4{addr: [192, 168, ..], port: _}) => None,
+										Some(NetAddress::IPv4{addr: [192, 88, 99, ..], port: _}) => None,
+										Some(NetAddress::IPv6{addr: [32..=63, ..], port: _}) => peer.their_net_address.clone(),
```

Comment:
Please use hex representation here so its clear what's up.

Reply:
Do you want me to use hex codes inside the `addr` array or replace the whole array with hex ?

Reply:
Just more readable as `0x20..=0x3f` is more readable given you're trying to write `2000::/3`. Also comments on each line noting what the CIDR is you're matching would be nice.